lib/string_decoder.js

```diff
@@ -282,3 +289,20 @@ function simpleWrite(buf) {
 function simpleEnd(buf) {
   return (buf && buf.length ? this.write(buf) : '');
 }
+
+function end(buf) {
+  let result = '';
+
+  if (this.encoding === 'utf16le')
+    result = utf16End.call(this, buf);
```

Comment:
If we're the only ones calling these methods directly, then we should be able to avoid the overhead of `.call()` and just pass an instance as another argument.

Reply:
Ack.